ui/apps/platform/src/Containers/MainPage/MainPage.tsx

```diff
@@ -98,6 +120,7 @@ function MainPage(): ReactElement {
                     hasServiceIdentityWritePermission={hasServiceIdentityWritePermission}
                 />
                 {metadata?.stale && <VersionOutOfDate />}
+                {!databaseAvailable && <DatabaseBanner />}
```

Comment:
Suggestion out of an abundance of caution, because polled request in `MainPage` element causes React to re-render the entire page (even though, in theory, reconciliation algorithm will decide no changes).

I am concerned if this new cause for re-rendering gives any descendant element that might have an unknown weakness not to re-render itself purely (that is, without side effects) an opening to cause subtle intermittent integration test failure because element detached from the DOM (which would be unrelated to test).

Therefore, what do you think if `DatabaseBanner` encapsulates such a frequent polled request. For what it is worth, that is the pattern in widgets on main dashboard and vulnerability management dashboard.

Sadly the precedents are equivocal:
* `UnreachableWarning` depends on `serverState` from Redux store via 'MainPage` but encapsulates `if return null`
* `Notifications` encapsulates its data access from Redux store
* `CredentialExpiryBanner` encapsulates data request
* `VersionOutOfDate` has conditional rendering which depends on `metadata?.state` from Redux store via `MainPage`